psm_utils/io/tsv.py

```diff
@@ -192,43 +199,44 @@ def __enter__(self) -> TSVWriter:
             self._writer.writeheader()
         return self
 
-    def __exit__(self, *args, **kwargs) -> None:
-        self._open_file.close()
-        self._open_file = None
+    def __exit__(self, *args: Any, **kwargs: Any) -> None:
+        """Exit context manager and clean up file resources."""
+        if self._open_file is not None:
```

Comment:
I'm kind of curious about the reason for this test. Did it mask an exception thrown in `__enter__`?

Reply:
Technically, it would be possible to call exit on the object outside of its context. Something that MyPy complains about, even though it's not a realistic issue.